### ISSUE TITLE
Incompatible with the new version of Keepass (2.57)

### ISSUE THREAD

**Issue by vorand** [sirAndros/KeePassWinHello#110]
**Describe the bug**

A few days ago, I installed a new version of Keepass, and as soon as I did so, the plugin stopped working. When I unlock my database, Keepass does not ask for my fingerprints anymore; instead, it presents me with a standard dialogue requesting my password.
The plugin has disappeared from the Plugins menu

**To Reproduce**

Download a portable version of Keepass 2.57 (64-bit)
Download the plugin.
Try to unlock a DB using fingerprints.

**Expected behavior**
1. DB is locked
2. When I am unlocking the DB, a Windows Hello window pops up and requests my fingerprints.

Please complete the following information:

| Component | Version |
|-----------|---------|
| Operating System |  Windows 11 22H2|
| KeePass |  2.57 (64-bit) portable|
| Plugin | 3.3.1 |

**Hardware info**

- Dell Lattitude 7400
- TPM module info: STM, version 1.258.0.0, Specification 2.0
  
 **Plugin settings**
 
 - [ ] Is persistent mode enabled ("Store keys in Windows Credential Manager" checked)
     Menu in not available any longer
 
 **KeePass settings and environment**
 
 - [x] Is secure desktop enabled
 - [ ] Is KeePass running under Administrator (in elevated process)
 - [ ] Was it right after hibernation
 - [ ] Was KeePass minified or closed

**Comment by MrFly72** [comment]
I moved back to 3.2 a while ago, as there was no reaction on the minimize problem of 3.3.1
With 3.2 my 2.57 seems to work.
Maybe you shoud check V3.2?

**Comment by vorand** [comment]
@MrFly72, thank you so much for the advice. Once I had rolled back to the version 3.2, the plugin started working again.
However, neither 3.3 nor 3.3.1 is working on my laptop.

**Comment by ilpssun** [comment]
I am using KeePass 2.57 (portable, 64bit) with WinHello 3.3.1.0 successfully.  Works well.  I have not used the plugin with prior versions, so I have no info about possible upgrade issues.  I'm on Windows 11 23H2.

**Comment by sambernet** [comment]
WinHello 3.3.1 works on my end with KeePass 2.57 (non-portable, installed through chocolatey) as well. But a colleague here is having issues with the slightly newer KeePass 2.57.1 and the very same WinHello version.

We are now testing combinations to gain more insights:
- downgrade to KeePass 2.57 (with WinHello 3.3.1)
- downgrade to WinHello 3.2 (on KeePass 2.57.1)
- upgrade from KeePass 2.57 to 2.57.1 on my machine (with WinHello 3.3.1, which is currently working with KeePass 2.57): ✅ Works just fine. So there's no general incompatibility obviously. OS version is Win 11 23H2 (Build 22631.4317)

**Comment by sambernet** [comment]
Turns out, just wiping the KeePass Plugin cache was all that was needed on his end (in KeePass use `Tools` -> `Plugins` -> `Clear Plugin Cache`, then fully restart KeePass).

**So all good now with KeePass `2.57.1` and WinHello `3.3.1`**... Sorry for the commotion.

@vorand and @MrFly72: maybe you could verify if that helps on your end as well - and close the issue in case it does :)

**Comment by MrFly72** [comment]
I just tried it again. So Winhello is working for me, but the minimze of the window is still there.
Basically on opening the file, the Winhello prompt will come up. When it finishes with face recognition, the Keepass Window is minimized and I can see, that it somehow decrypts, as the icon in the task-bar will show activity. But keepass will stay minimized, which is kind of quirky, as the intention of a user is, to open the keepass container, right ?
So basically I have to do this with 3.3.1:
1. Open Keepass
2. Open database
3. Finish Hello Prompt (either pin or face)
4. leave the icon in the tray until the "working bar" finishes
5. "reopen" keepass from the task tray.
Basically this is my issue:
https://github.com/sirAndros/KeePassWinHello/issues/107

**Comment by sambernet** [comment]
Hmm true, I never really noticed this only happens when using WinHello. I just tested with WinHello disabled, and it doesn't happen. Note there's actually a specific setting for that behavior in KeePass itself (Tools -> Options -> Interface -> Minimize main window after opening a database), but I see that I have it disabled.

I just tested with the WinHello plugin disabled, and as you say, the setting in the options then works as intended, that is, the KeePass window is left open after unlock (or not, depending on what is set). 
In contrast, with WinHello enabled, it is always minimized after unlocking. Checking/Unchecking the relevent KeePass setting with WinHello enabled doesn't fix it either.

I don't actually care about that a lot, as I mostly use auto-type and also have KeePassRPC with the matching browser plugins installed - so I more often need to just unlock the KeePass safe and have it minimized anyway, than actually open it and manually copy some credentials.

I would suggest opening a new/separate ticket for this if it bothers you, as it's quite a different topic than what this issue is about 😉

**Comment by MrFly72** [comment]
There is already one open (see my post above your last one with a link to it). 
But so far nobody was interested in it.
